// @wordpress/edit-post/build/components/secondary-sidebar/inserter-sidebar.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = InserterSidebar;

var _element = require("@wordpress/element");

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _data = require("@wordpress/data");

var _components = require("@wordpress/components");

var _blockEditor = require("@wordpress/block-editor");

var _icons = require("@wordpress/icons");

var _compose = require("@wordpress/compose");

var _store = require("../../store");

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
function InserterSidebar() {
  const {
    insertionPoint,
    showMostUsedBlocks
  } = (0, _data.useSelect)(select => {
    const {
      isFeatureActive,
      __experimentalGetInsertionPoint
    } = select(_store.store);
    return {
      insertionPoint: __experimentalGetInsertionPoint(),
      showMostUsedBlocks: isFeatureActive('mostUsedBlocks')
    };
  }, []);
  const {
    setIsInserterOpened
  } = (0, _data.useDispatch)(_store.store);
  const isMobileViewport = (0, _compose.useViewportMatch)('medium', '<');
  const [inserterDialogRef, inserterDialogProps] = (0, _compose.__experimentalUseDialog)({
    onClose: () => setIsInserterOpened(false)
  });
  return (0, _element.createElement)("div", (0, _extends2.default)({
    ref: inserterDialogRef
  }, inserterDialogProps, {
    className: "edit-post-editor__inserter-panel"
  }), (0, _element.createElement)("div", {
    className: "edit-post-editor__inserter-panel-header"
  }, (0, _element.createElement)(_components.Button, {
    icon: _icons.close,
    onClick: () => setIsInserterOpened(false)
  })), (0, _element.createElement)("div", {
    className: "edit-post-editor__inserter-panel-content"
  }, (0, _element.createElement)(_blockEditor.__experimentalLibrary, {
    showMostUsedBlocks: showMostUsedBlocks,
    showInserterHelpPanel: true,
    shouldFocusBlock: isMobileViewport,
    rootClientId: insertionPoint.rootClientId,
    __experimentalInsertionIndex: insertionPoint.insertionIndex,
    __experimentalFilterValue: insertionPoint.filterValue
  })));
}
