// @wordpress/edit-post/build/components/secondary-sidebar/list-view-sidebar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ListViewSidebar;

var _element = require("@wordpress/element");

var _blockEditor = require("@wordpress/block-editor");

var _components = require("@wordpress/components");

var _compose = require("@wordpress/compose");

var _data = require("@wordpress/data");

var _i18n = require("@wordpress/i18n");

var _icons = require("@wordpress/icons");

var _keycodes = require("@wordpress/keycodes");

var _store = require("../../store");

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
function ListViewSidebar() {
  const {
    setIsListViewOpened
  } = (0, _data.useDispatch)(_store.store);
  const {
    clearSelectedBlock,
    selectBlock
  } = (0, _data.useDispatch)(_blockEditor.store);

  async function selectEditorBlock(clientId) {
    await clearSelectedBlock();
    selectBlock(clientId, -1);
  }

  const focusOnMountRef = (0, _compose.useFocusOnMount)('firstElement');
  const focusReturnRef = (0, _compose.useFocusReturn)();

  function closeOnEscape(event) {
    if (event.keyCode === _keycodes.ESCAPE && !event.defaultPrevented) {
      event.preventDefault();
      setIsListViewOpened(false);
    }
  }

  const instanceId = (0, _compose.useInstanceId)(ListViewSidebar);
  const labelId = `edit-post-editor__list-view-panel-label-${instanceId}`;
  return (// eslint-disable-next-line jsx-a11y/no-static-element-interactions
    (0, _element.createElement)("div", {
      "aria-labelledby": labelId,
      className: "edit-post-editor__list-view-panel",
      onKeyDown: closeOnEscape
    }, (0, _element.createElement)("div", {
      className: "edit-post-editor__list-view-panel-header"
    }, (0, _element.createElement)("strong", {
      id: labelId
    }, (0, _i18n.__)('List view')), (0, _element.createElement)(_components.Button, {
      icon: _icons.closeSmall,
      label: (0, _i18n.__)('Close list view sidebar'),
      onClick: () => setIsListViewOpened(false)
    })), (0, _element.createElement)("div", {
      className: "edit-post-editor__list-view-panel-content",
      ref: (0, _compose.useMergeRefs)([focusReturnRef, focusOnMountRef])
    }, (0, _element.createElement)(_blockEditor.__experimentalListView, {
      onSelect: selectEditorBlock,
      showNestedBlocks: true,
      __experimentalPersistentListViewFeatures: true
    })))
  );
}
