// @wordpress/editor/build/components/index.native.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  AutosaveMonitor: true,
  PostTitle: true,
  EditorHistoryRedo: true,
  EditorHistoryUndo: true,
  EditorProvider: true,
  EditorHelpTopics: true
};
Object.defineProperty(exports, "AutosaveMonitor", {
  enumerable: true,
  get: function () {
    return _autosaveMonitor.default;
  }
});
Object.defineProperty(exports, "PostTitle", {
  enumerable: true,
  get: function () {
    return _postTitle.default;
  }
});
Object.defineProperty(exports, "EditorHistoryRedo", {
  enumerable: true,
  get: function () {
    return _redo.default;
  }
});
Object.defineProperty(exports, "EditorHistoryUndo", {
  enumerable: true,
  get: function () {
    return _undo.default;
  }
});
Object.defineProperty(exports, "EditorProvider", {
  enumerable: true,
  get: function () {
    return _provider.default;
  }
});
Object.defineProperty(exports, "EditorHelpTopics", {
  enumerable: true,
  get: function () {
    return _editorHelp.default;
  }
});

var _autosaveMonitor = _interopRequireDefault(require("./autosave-monitor"));

var _postTitle = _interopRequireDefault(require("./post-title"));

var _redo = _interopRequireDefault(require("./editor-history/redo"));

var _undo = _interopRequireDefault(require("./editor-history/undo"));

var _provider = _interopRequireDefault(require("./provider"));

var _editorHelp = _interopRequireDefault(require("./editor-help"));

var _deprecated = require("./deprecated");

Object.keys(_deprecated).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _deprecated[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _deprecated[key];
    }
  });
});
