// @wordpress/block-library/build/gallery/v1/update-gallery-modal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = UpdateGalleryModal;
exports.updateGallery = void 0;

var _element = require("@wordpress/element");

var _components = require("@wordpress/components");

var _i18n = require("@wordpress/i18n");

var _data = require("@wordpress/data");

var _blocks = require("@wordpress/blocks");

var _blockEditor = require("@wordpress/block-editor");

var _constants = require("./constants");

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
const updateGallery = ({
  clientId,
  getBlock,
  replaceBlocks
}) => () => {
  let link;
  const {
    attributes: {
      sizeSlug,
      linkTo,
      images,
      caption
    }
  } = getBlock(clientId);

  switch (linkTo) {
    case 'post':
      link = _constants.LINK_DESTINATION_ATTACHMENT;
      break;

    case 'file':
      link = _constants.LINK_DESTINATION_MEDIA;
      break;

    default:
      link = _constants.LINK_DESTINATION_NONE;
      break;
  }

  const innerBlocks = images.map(image => (0, _blocks.createBlock)('core/image', {
    id: parseInt(image.id, 10),
    url: image.url,
    alt: image.alt,
    caption: image.caption,
    linkDestination: link
  }));
  replaceBlocks(clientId, (0, _blocks.createBlock)('core/gallery', {
    sizeSlug,
    linkTo: link,
    caption
  }, innerBlocks));
};

exports.updateGallery = updateGallery;

function UpdateGalleryModal({
  onClose,
  clientId
}) {
  const {
    getBlock
  } = (0, _data.useSelect)(_blockEditor.store);
  const {
    replaceBlocks
  } = (0, _data.useDispatch)(_blockEditor.store);
  return (0, _element.createElement)(_components.Modal, {
    closeLabel: (0, _i18n.__)('Close'),
    onRequestClose: onClose,
    title: (0, _i18n.__)('Update gallery'),
    className: 'wp-block-update-gallery-modal',
    aria: {
      describedby: 'wp-block-update-gallery-modal__description'
    }
  }, (0, _element.createElement)("p", {
    id: 'wp-block-update-gallery-modal__description'
  }, (0, _i18n.__)('Updating to the new format adds the ability to use custom links or styles on individual images in the gallery, and makes it easier to add or move them around.')), (0, _element.createElement)("div", {
    className: "wp-block-update-gallery-modal-buttons"
  }, (0, _element.createElement)(_components.Button, {
    isTertiary: true,
    onClick: onClose
  }, (0, _i18n.__)('Cancel')), (0, _element.createElement)(_components.Button, {
    isPrimary: true,
    onClick: updateGallery({
      replaceBlocks,
      getBlock,
      clientId,
      createBlock: _blocks.createBlock
    })
  }, (0, _i18n.__)('Update'))));
}
