// @wordpress/edit-post/build/components/header/index.native.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _element = require("@wordpress/element");

var _reactNative = require("react-native");

require("@wordpress/interface");

var _headerToolbar = _interopRequireDefault(require("./header-toolbar"));

/**
 * External dependencies
 */

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
class Header extends _element.Component {
  constructor() {
    super(...arguments);
    this.keyboardDidShow = this.keyboardDidShow.bind(this);
    this.keyboardDidHide = this.keyboardDidHide.bind(this);
    this.state = {
      isKeyboardVisible: false
    };
  }

  componentDidMount() {
    _reactNative.Keyboard.addListener('keyboardDidShow', this.keyboardDidShow);

    _reactNative.Keyboard.addListener('keyboardDidHide', this.keyboardDidHide);
  }

  componentWillUnmount() {
    _reactNative.Keyboard.removeListener('keyboardDidShow', this.keyboardDidShow);

    _reactNative.Keyboard.removeListener('keyboardDidHide', this.keyboardDidHide);
  }

  keyboardDidShow() {
    this.setState({
      isKeyboardVisible: true
    });
  }

  keyboardDidHide() {
    this.setState({
      isKeyboardVisible: false
    });
  }

  render() {
    return (0, _element.createElement)(_headerToolbar.default, {
      showKeyboardHideButton: this.state.isKeyboardVisible
    });
  }

}

exports.default = Header;
