// @wordpress/block-library/build/buttons/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.settings = exports.name = exports.metadata = void 0;

var _i18n = require("@wordpress/i18n");

var _icons = require("@wordpress/icons");

var _deprecated = _interopRequireDefault(require("./deprecated"));

var _transforms = _interopRequireDefault(require("./transforms"));

var _edit = _interopRequireDefault(require("./edit"));

var _save = _interopRequireDefault(require("./save"));

var _variations = _interopRequireDefault(require("./variations"));

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
const metadata = {
  apiVersion: 2,
  name: "core/buttons",
  title: "Buttons",
  category: "design",
  description: "Prompt visitors to take action with a group of button-style links.",
  keywords: ["link"],
  textdomain: "default",
  attributes: {
    contentJustification: {
      type: "string"
    },
    orientation: {
      type: "string",
      "default": "horizontal"
    }
  },
  supports: {
    anchor: true,
    align: ["wide", "full"],
    __experimentalExposeControlsToChildren: true
  },
  editorStyle: "wp-block-buttons-editor",
  style: "wp-block-buttons"
};
exports.metadata = metadata;
const {
  name
} = metadata;
exports.name = name;
const settings = {
  icon: _icons.buttons,
  example: {
    innerBlocks: [{
      name: 'core/button',
      attributes: {
        text: (0, _i18n.__)('Find out more')
      }
    }, {
      name: 'core/button',
      attributes: {
        text: (0, _i18n.__)('Contact us')
      }
    }]
  },
  deprecated: _deprecated.default,
  transforms: _transforms.default,
  edit: _edit.default,
  save: _save.default,
  variations: _variations.default
};
exports.settings = settings;
