// @wordpress/components/build/ui/color-picker/input-with-slider.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputWithSlider = void 0;

var _element = require("@wordpress/element");

var _hStack = require("../../h-stack");

var _text = require("../../text");

var _spacer = require("../../spacer");

var _space = require("../utils/space");

var _styles = require("./styles");

/**
 * Internal dependencies
 */
const InputWithSlider = ({
  min,
  max,
  label,
  abbreviation,
  onChange,
  value
}) => {
  return (0, _element.createElement)(_spacer.Spacer, {
    as: _hStack.HStack,
    spacing: 4
  }, (0, _element.createElement)(_styles.NumberControlWrapper, {
    min: min,
    max: max,
    label: label,
    hideLabelFromVision: true,
    value: value,
    onChange: onChange,
    prefix: (0, _element.createElement)(_spacer.Spacer, {
      as: _text.Text,
      paddingLeft: (0, _space.space)(3.5),
      color: "blue",
      lineHeight: 1
    }, abbreviation),
    hideHTMLArrows: true
  }), (0, _element.createElement)(_styles.RangeControl, {
    label: label,
    hideLabelFromVision: true,
    min: min,
    max: max,
    value: value,
    onChange: onChange,
    withInputField: false
  }));
};

exports.InputWithSlider = InputWithSlider;
