// @wordpress/block-editor/build/components/inserter/reusable-blocks-tab.native.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _element = require("@wordpress/element");

var _data = require("@wordpress/data");

var _blockTypesList = _interopRequireDefault(require("../block-types-list"));

var _store = require("../../store");

var _utils = require("./utils");

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
function ReusableBlocksTab({
  onSelect,
  rootClientId,
  listProps
}) {
  const {
    items
  } = (0, _data.useSelect)(select => {
    const {
      getInserterItems
    } = select(_store.store);
    const allItems = getInserterItems(rootClientId);
    return {
      items: (0, _utils.filterInserterItems)(allItems, {
        onlyReusable: true
      })
    };
  }, [rootClientId]);
  return (0, _element.createElement)(_blockTypesList.default, {
    name: "ReusableBlocks",
    items: items,
    onSelect: onSelect,
    listProps: listProps
  });
}

var _default = ReusableBlocksTab;
exports.default = _default;
