// @wordpress/editor/build/store/defaults.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EDITOR_SETTINGS_DEFAULTS = exports.PREFERENCES_DEFAULTS = void 0;

var _blockEditor = require("@wordpress/block-editor");

/**
 * WordPress dependencies
 */
const PREFERENCES_DEFAULTS = {
  insertUsage: {},
  // Should be kept for backward compatibility, see: https://github.com/WordPress/gutenberg/issues/14580.
  isPublishSidebarEnabled: true
};
/**
 * The default post editor settings
 *
 *  allowedBlockTypes  boolean|Array Allowed block types
 *  richEditingEnabled boolean       Whether rich editing is enabled or not
 *  codeEditingEnabled boolean       Whether code editing is enabled or not
 *  enableCustomFields boolean       Whether the WordPress custom fields are enabled or not.
 *                                     true  = the user has opted to show the Custom Fields panel at the bottom of the editor.
 *                                     false = the user has opted to hide the Custom Fields panel at the bottom of the editor.
 *                                     undefined = the current environment does not support Custom Fields,
 *                                                 so the option toggle in Preferences -> Panels to
 *                                                 enable the Custom Fields panel is not displayed.
 *  autosaveInterval   number        Autosave Interval
 *  availableTemplates array?        The available post templates
 *  disablePostFormats boolean       Whether or not the post formats are disabled
 *  allowedMimeTypes   array?        List of allowed mime types and file extensions
 *  maxUploadFileSize  number        Maximum upload file size
 *  supportsLayout     boolean      Whether the editor supports layouts.
 */

exports.PREFERENCES_DEFAULTS = PREFERENCES_DEFAULTS;
const EDITOR_SETTINGS_DEFAULTS = { ..._blockEditor.SETTINGS_DEFAULTS,
  richEditingEnabled: true,
  codeEditingEnabled: true,
  enableCustomFields: undefined,
  supportsLayout: true
};
exports.EDITOR_SETTINGS_DEFAULTS = EDITOR_SETTINGS_DEFAULTS;
