// @wordpress/edit-post/build-module/components/header/more-menu/index.js
import { createElement, Fragment } from "@wordpress/element";

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { MenuGroup } from '@wordpress/components';
import { ActionItem, MoreMenuDropdown, PinnedItems } from '@wordpress/interface';
import { useViewportMatch } from '@wordpress/compose';
/**
 * Internal dependencies
 */

import ModeSwitcher from '../mode-switcher';
import PreferencesMenuItem from '../preferences-menu-item';
import ToolsMoreMenuGroup from '../tools-more-menu-group';
import WritingMenu from '../writing-menu';
const POPOVER_PROPS = {
  className: 'edit-post-more-menu__content'
};

const MoreMenu = ({
  showIconLabels
}) => {
  const isLargeViewport = useViewportMatch('large');
  return createElement(MoreMenuDropdown, {
    className: "edit-post-more-menu",
    popoverProps: POPOVER_PROPS,
    toggleProps: {
      showTooltip: !showIconLabels,
      ...(showIconLabels && {
        variant: 'tertiary'
      })
    }
  }, ({
    onClose
  }) => createElement(Fragment, null, showIconLabels && !isLargeViewport && createElement(PinnedItems.Slot, {
    className: showIconLabels && 'show-icon-labels',
    scope: "core/edit-post"
  }), createElement(WritingMenu, null), createElement(ModeSwitcher, null), createElement(ActionItem.Slot, {
    name: "core/edit-post/plugin-more-menu",
    label: __('Plugins'),
    as: MenuGroup,
    fillProps: {
      onClick: onClose
    }
  }), createElement(ToolsMoreMenuGroup.Slot, {
    fillProps: {
      onClose
    }
  }), createElement(MenuGroup, null, createElement(PreferencesMenuItem, null))));
};

export default MoreMenu;
