// @wordpress/block-library/build-module/buttons/index.js
/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { buttons as icon } from '@wordpress/icons';
/**
 * Internal dependencies
 */

import deprecated from './deprecated';
import transforms from './transforms';
import edit from './edit';
const metadata = {
  apiVersion: 2,
  name: "core/buttons",
  title: "Buttons",
  category: "design",
  description: "Prompt visitors to take action with a group of button-style links.",
  keywords: ["link"],
  textdomain: "default",
  attributes: {
    contentJustification: {
      type: "string"
    },
    orientation: {
      type: "string",
      "default": "horizontal"
    }
  },
  supports: {
    anchor: true,
    align: ["wide", "full"],
    __experimentalExposeControlsToChildren: true
  },
  editorStyle: "wp-block-buttons-editor",
  style: "wp-block-buttons"
};
import save from './save';
import variations from './variations';
const {
  name
} = metadata;
export { metadata, name };
export const settings = {
  icon,
  example: {
    innerBlocks: [{
      name: 'core/button',
      attributes: {
        text: __('Find out more')
      }
    }, {
      name: 'core/button',
      attributes: {
        text: __('Contact us')
      }
    }]
  },
  deprecated,
  transforms,
  edit,
  save,
  variations
};
