// @wordpress/edit-post/build/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.reinitializeEditor = reinitializeEditor;
exports.initializeEditor = initializeEditor;
Object.defineProperty(exports, "PluginBlockSettingsMenuItem", {
  enumerable: true,
  get: function () {
    return _pluginBlockSettingsMenuItem.default;
  }
});
Object.defineProperty(exports, "PluginDocumentSettingPanel", {
  enumerable: true,
  get: function () {
    return _pluginDocumentSettingPanel.default;
  }
});
Object.defineProperty(exports, "PluginMoreMenuItem", {
  enumerable: true,
  get: function () {
    return _pluginMoreMenuItem.default;
  }
});
Object.defineProperty(exports, "PluginPostPublishPanel", {
  enumerable: true,
  get: function () {
    return _pluginPostPublishPanel.default;
  }
});
Object.defineProperty(exports, "PluginPostStatusInfo", {
  enumerable: true,
  get: function () {
    return _pluginPostStatusInfo.default;
  }
});
Object.defineProperty(exports, "PluginPrePublishPanel", {
  enumerable: true,
  get: function () {
    return _pluginPrePublishPanel.default;
  }
});
Object.defineProperty(exports, "PluginSidebar", {
  enumerable: true,
  get: function () {
    return _pluginSidebar.default;
  }
});
Object.defineProperty(exports, "PluginSidebarMoreMenuItem", {
  enumerable: true,
  get: function () {
    return _pluginSidebarMoreMenuItem.default;
  }
});
Object.defineProperty(exports, "__experimentalFullscreenModeClose", {
  enumerable: true,
  get: function () {
    return _fullscreenModeClose.default;
  }
});
Object.defineProperty(exports, "__experimentalMainDashboardButton", {
  enumerable: true,
  get: function () {
    return _mainDashboardButton.default;
  }
});
Object.defineProperty(exports, "store", {
  enumerable: true,
  get: function () {
    return _store.store;
  }
});

var _element = require("@wordpress/element");

var _blockLibrary = require("@wordpress/block-library");

var _data = require("@wordpress/data");

var _interface = require("@wordpress/interface");

require("./hooks");

require("./plugins");

var _editor = _interopRequireDefault(require("./editor"));

var _pluginBlockSettingsMenuItem = _interopRequireDefault(require("./components/block-settings-menu/plugin-block-settings-menu-item"));

var _pluginDocumentSettingPanel = _interopRequireDefault(require("./components/sidebar/plugin-document-setting-panel"));

var _pluginMoreMenuItem = _interopRequireDefault(require("./components/header/plugin-more-menu-item"));

var _pluginPostPublishPanel = _interopRequireDefault(require("./components/sidebar/plugin-post-publish-panel"));

var _pluginPostStatusInfo = _interopRequireDefault(require("./components/sidebar/plugin-post-status-info"));

var _pluginPrePublishPanel = _interopRequireDefault(require("./components/sidebar/plugin-pre-publish-panel"));

var _pluginSidebar = _interopRequireDefault(require("./components/sidebar/plugin-sidebar"));

var _pluginSidebarMoreMenuItem = _interopRequireDefault(require("./components/header/plugin-sidebar-more-menu-item"));

var _fullscreenModeClose = _interopRequireDefault(require("./components/header/fullscreen-mode-close"));

var _mainDashboardButton = _interopRequireDefault(require("./components/header/main-dashboard-button"));

var _store = require("./store");

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */

/**
 * Reinitializes the editor after the user chooses to reboot the editor after
 * an unhandled error occurs, replacing previously mounted editor element using
 * an initial state from prior to the crash.
 *
 * @param {Object}  postType     Post type of the post to edit.
 * @param {Object}  postId       ID of the post to edit.
 * @param {Element} target       DOM node in which editor is rendered.
 * @param {?Object} settings     Editor settings object.
 * @param {Object}  initialEdits Programmatic edits to apply initially, to be
 *                               considered as non-user-initiated (bypass for
 *                               unsaved changes prompt).
 */
function reinitializeEditor(postType, postId, target, settings, initialEdits) {
  (0, _element.unmountComponentAtNode)(target);
  const reboot = reinitializeEditor.bind(null, postType, postId, target, settings, initialEdits);
  (0, _element.render)((0, _element.createElement)(_editor.default, {
    settings: settings,
    onError: reboot,
    postId: postId,
    postType: postType,
    initialEdits: initialEdits,
    recovery: true
  }), target);
}
/**
 * Initializes and returns an instance of Editor.
 *
 * @param {string}  id           Unique identifier for editor instance.
 * @param {string}  postType     Post type of the post to edit.
 * @param {Object}  postId       ID of the post to edit.
 * @param {?Object} settings     Editor settings object.
 * @param {Object}  initialEdits Programmatic edits to apply initially, to be
 *                               considered as non-user-initiated (bypass for
 *                               unsaved changes prompt).
 */


function initializeEditor(id, postType, postId, settings, initialEdits) {
  const target = document.getElementById(id);
  const reboot = reinitializeEditor.bind(null, postType, postId, target, settings, initialEdits);
  (0, _data.dispatch)(_interface.store).setFeatureDefaults('core/edit-post', {
    fixedToolbar: false,
    welcomeGuide: true,
    fullscreenMode: true,
    showIconLabels: false,
    themeStyles: true,
    showBlockBreadcrumbs: true,
    welcomeGuideTemplate: true
  });
  (0, _blockLibrary.registerCoreBlocks)();

  if (process.env.GUTENBERG_PHASE === 2) {
    (0, _blockLibrary.__experimentalRegisterExperimentalCoreBlocks)({
      enableFSEBlocks: settings.__unstableEnableFullSiteEditingBlocks
    });
  } // Show a console log warning if the browser is not in Standards rendering mode.


  const documentMode = document.compatMode === 'CSS1Compat' ? 'Standards' : 'Quirks';

  if (documentMode !== 'Standards') {
    // eslint-disable-next-line no-console
    console.warn("Your browser is using Quirks Mode. \nThis can cause rendering issues such as blocks overlaying meta boxes in the editor. Quirks Mode can be triggered by PHP errors or HTML code appearing before the opening <!DOCTYPE html>. Try checking the raw page source or your site's PHP error log and resolving errors there, removing any HTML before the doctype, or disabling plugins.");
  } // This is a temporary fix for a couple of issues specific to Webkit on iOS.
  // Without this hack the browser scrolls the mobile toolbar off-screen.
  // Once supported in Safari we can replace this in favor of preventScroll.
  // For details see issue #18632 and PR #18686
  // Specifically, we scroll `interface-interface-skeleton__body` to enable a fixed top toolbar.
  // But Mobile Safari forces the `html` element to scroll upwards, hiding the toolbar.


  const isIphone = window.navigator.userAgent.indexOf('iPhone') !== -1;

  if (isIphone) {
    window.addEventListener('scroll', event => {
      const editorScrollContainer = document.getElementsByClassName('interface-interface-skeleton__body')[0];

      if (event.target === document) {
        // Scroll element into view by scrolling the editor container by the same amount
        // that Mobile Safari tried to scroll the html element upwards.
        if (window.scrollY > 100) {
          editorScrollContainer.scrollTop = editorScrollContainer.scrollTop + window.scrollY;
        } // Undo unwanted scroll on html element, but only in the visual editor.


        if (document.getElementsByClassName('is-mode-visual')[0]) {
          window.scrollTo(0, 0);
        }
      }
    });
  }

  (0, _element.render)((0, _element.createElement)(_editor.default, {
    settings: settings,
    onError: reboot,
    postId: postId,
    postType: postType,
    initialEdits: initialEdits
  }), target);
}
