// @wordpress/block-library/build/template-part/edit/selection/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = TemplatePartSelection;

var _element = require("@wordpress/element");

var _components = require("@wordpress/components");

var _templatePartPreviews = _interopRequireDefault(require("./template-part-previews"));

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
function TemplatePartSelection({
  setAttributes,
  onClose,
  area,
  templatePartId = null
}) {
  const [filterValue, setFilterValue] = (0, _element.useState)('');
  return (0, _element.createElement)("div", null, (0, _element.createElement)(_components.SearchControl, {
    value: filterValue,
    onChange: setFilterValue,
    className: "wp-block-template-part__selection-preview-search-form"
  }), (0, _element.createElement)("div", {
    className: "wp-block-template-part__selection-preview-container"
  }, (0, _element.createElement)(_templatePartPreviews.default, {
    setAttributes: setAttributes,
    filterValue: filterValue,
    onClose: onClose,
    area: area,
    templatePartId: templatePartId
  })));
}
