// @wordpress/block-library/build/post-navigation-link/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.settings = exports.name = exports.metadata = void 0;

var _edit = _interopRequireDefault(require("./edit"));

var _variations = _interopRequireDefault(require("./variations"));

/**
 * Internal dependencies
 */
const metadata = {
  apiVersion: 2,
  name: "core/post-navigation-link",
  title: "Post Navigation Link",
  category: "theme",
  description: "Displays the next or previous post link that is adjacent to the current post.",
  textdomain: "default",
  attributes: {
    textAlign: {
      type: "string"
    },
    type: {
      type: "string",
      "default": "next"
    },
    label: {
      type: "string"
    },
    showTitle: {
      type: "boolean",
      "default": false
    },
    linkLabel: {
      type: "boolean",
      "default": false
    }
  },
  supports: {
    reusable: false,
    html: false,
    typography: {
      fontSize: true,
      lineHeight: true
    }
  }
};
exports.metadata = metadata;
const {
  name
} = metadata;
exports.name = name;
const settings = {
  edit: _edit.default,
  variations: _variations.default
};
exports.settings = settings;
