// @wordpress/edit-post/build-module/components/sidebar/post-schedule/index.js
import { createElement, Fragment } from "@wordpress/element";

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { PanelRow, Dropdown, Button } from '@wordpress/components';
import { useRef } from '@wordpress/element';
import { PostSchedule as PostScheduleForm, PostScheduleLabel, PostScheduleCheck } from '@wordpress/editor';
export function PostSchedule() {
  const anchorRef = useRef();
  return createElement(PostScheduleCheck, null, createElement(PanelRow, {
    className: "edit-post-post-schedule",
    ref: anchorRef
  }, createElement("span", null, __('Publish')), createElement(Dropdown, {
    popoverProps: {
      anchorRef: anchorRef.current
    },
    position: "bottom left",
    contentClassName: "edit-post-post-schedule__dialog",
    renderToggle: ({
      onToggle,
      isOpen
    }) => createElement(Fragment, null, createElement(Button, {
      className: "edit-post-post-schedule__toggle",
      onClick: onToggle,
      "aria-expanded": isOpen,
      variant: "tertiary"
    }, createElement(PostScheduleLabel, null))),
    renderContent: () => createElement(PostScheduleForm, null)
  })));
}
export default PostSchedule;
