// @wordpress/block-library/build/template-part/edit/placeholder/patterns-setup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = PatternsSetup;

var _element = require("@wordpress/element");

var _blockEditor = require("@wordpress/block-editor");

var _i18n = require("@wordpress/i18n");

var _components = require("@wordpress/components");

/**
 * WordPress dependencies
 */
function PatternsSetup({
  area,
  areaLabel,
  areaIcon,
  clientId,
  onCreate,
  resetPlaceholder
}) {
  const blockNameWithArea = area ? `core/template-part/${area}` : 'core/template-part'; // Restructure onCreate to set the blocks on local state.
  // Add modal to confirm title and trigger onCreate.

  const [title, setTitle] = (0, _element.useState)((0, _i18n.__)('Untitled Template Part'));
  const [startingBlocks, setStartingBlocks] = (0, _element.useState)([]);
  const [isTitleStep, setIsTitleStep] = (0, _element.useState)(false);

  const selectPattern = selectedPattern => {
    setStartingBlocks(selectedPattern);
    setIsTitleStep(true);
  };

  const submitForCreation = event => {
    event.preventDefault();
    onCreate(startingBlocks, title);
  };

  return (0, _element.createElement)(_element.Fragment, null, (0, _element.createElement)(_blockEditor.__experimentalBlockPatternSetup, {
    clientId: clientId,
    startBlankComponent: (0, _element.createElement)(StartBlankComponent, {
      setTitleStep: setIsTitleStep,
      areaLabel: areaLabel,
      areaIcon: areaIcon
    }),
    onBlockPatternSelect: selectPattern,
    filterPatternsFn: pattern => {
      var _pattern$blockTypes, _pattern$blockTypes$s;

      return pattern === null || pattern === void 0 ? void 0 : (_pattern$blockTypes = pattern.blockTypes) === null || _pattern$blockTypes === void 0 ? void 0 : (_pattern$blockTypes$s = _pattern$blockTypes.some) === null || _pattern$blockTypes$s === void 0 ? void 0 : _pattern$blockTypes$s.call(_pattern$blockTypes, blockType => blockType === blockNameWithArea);
    }
  }), isTitleStep && (0, _element.createElement)(_components.Modal, {
    title: (0, _i18n.sprintf)('Name and create your new %s', areaLabel.toLowerCase()),
    closeLabel: (0, _i18n.__)('Cancel'),
    onRequestClose: resetPlaceholder,
    overlayClassName: "wp-block-template-part__placeholder-create-new__title-form"
  }, (0, _element.createElement)("form", {
    onSubmit: submitForCreation
  }, (0, _element.createElement)(_components.TextControl, {
    label: (0, _i18n.__)('Name'),
    value: title,
    onChange: setTitle
  }), (0, _element.createElement)(_components.Flex, {
    className: "wp-block-template-part__placeholder-create-new__title-form-actions",
    justify: "flex-end"
  }, (0, _element.createElement)(_components.FlexItem, null, (0, _element.createElement)(_components.Button, {
    variant: "secondary",
    onClick: resetPlaceholder
  }, (0, _i18n.__)('Cancel'))), (0, _element.createElement)(_components.FlexItem, null, (0, _element.createElement)(_components.Button, {
    variant: "primary",
    type: "submit",
    disabled: !title.length,
    "aria-disabled": !title.length
  }, (0, _i18n.__)('Create')))))));
}

function StartBlankComponent({
  setTitleStep,
  areaLabel,
  areaIcon
}) {
  (0, _element.useEffect)(() => {
    setTitleStep(true);
  }, []);
  return (0, _element.createElement)(_components.Placeholder, {
    label: areaLabel,
    icon: areaIcon,
    instructions: (0, _i18n.sprintf)( // Translators: %s as template part area title ("Header", "Footer", "Template Part", etc.).
    'Creating your new %s...', areaLabel.toLowerCase())
  });
}
