// @wordpress/edit-post/build/components/welcome-guide/default.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = WelcomeGuideDefault;

var _element = require("@wordpress/element");

var _data = require("@wordpress/data");

var _components = require("@wordpress/components");

var _i18n = require("@wordpress/i18n");

var _image = _interopRequireDefault(require("./image"));

var _store = require("../../store");

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
function WelcomeGuideDefault() {
  const {
    toggleFeature
  } = (0, _data.useDispatch)(_store.store);
  return (0, _element.createElement)(_components.Guide, {
    className: "edit-post-welcome-guide",
    contentLabel: (0, _i18n.__)('Welcome to the block editor'),
    finishButtonText: (0, _i18n.__)('Get started'),
    onFinish: () => toggleFeature('welcomeGuide'),
    pages: [{
      image: (0, _element.createElement)(_image.default, {
        nonAnimatedSrc: "https://s.w.org/images/block-editor/welcome-canvas.svg",
        animatedSrc: "https://s.w.org/images/block-editor/welcome-canvas.gif"
      }),
      content: (0, _element.createElement)(_element.Fragment, null, (0, _element.createElement)("h1", {
        className: "edit-post-welcome-guide__heading"
      }, (0, _i18n.__)('Welcome to the block editor')), (0, _element.createElement)("p", {
        className: "edit-post-welcome-guide__text"
      }, (0, _i18n.__)('In the WordPress editor, each paragraph, image, or video is presented as a distinct “block” of content.')))
    }, {
      image: (0, _element.createElement)(_image.default, {
        nonAnimatedSrc: "https://s.w.org/images/block-editor/welcome-editor.svg",
        animatedSrc: "https://s.w.org/images/block-editor/welcome-editor.gif"
      }),
      content: (0, _element.createElement)(_element.Fragment, null, (0, _element.createElement)("h1", {
        className: "edit-post-welcome-guide__heading"
      }, (0, _i18n.__)('Make each block your own')), (0, _element.createElement)("p", {
        className: "edit-post-welcome-guide__text"
      }, (0, _i18n.__)('Each block comes with its own set of controls for changing things like color, width, and alignment. These will show and hide automatically when you have a block selected.')))
    }, {
      image: (0, _element.createElement)(_image.default, {
        nonAnimatedSrc: "https://s.w.org/images/block-editor/welcome-library.svg",
        animatedSrc: "https://s.w.org/images/block-editor/welcome-library.gif"
      }),
      content: (0, _element.createElement)(_element.Fragment, null, (0, _element.createElement)("h1", {
        className: "edit-post-welcome-guide__heading"
      }, (0, _i18n.__)('Get to know the block library')), (0, _element.createElement)("p", {
        className: "edit-post-welcome-guide__text"
      }, (0, _element.createInterpolateElement)((0, _i18n.__)('All of the blocks available to you live in the block library. You’ll find it wherever you see the <InserterIconImage /> icon.'), {
        InserterIconImage: (0, _element.createElement)("img", {
          alt: (0, _i18n.__)('inserter'),
          src: "data:image/svg+xml,%3Csvg width='18' height='18' viewBox='0 0 18 18' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Crect width='18' height='18' rx='2' fill='%231E1E1E'/%3E%3Cpath d='M9.22727 4V14M4 8.77273H14' stroke='white' stroke-width='1.5'/%3E%3C/svg%3E%0A"
        })
      })))
    }, {
      image: (0, _element.createElement)(_image.default, {
        nonAnimatedSrc: "https://s.w.org/images/block-editor/welcome-documentation.svg",
        animatedSrc: "https://s.w.org/images/block-editor/welcome-documentation.gif"
      }),
      content: (0, _element.createElement)(_element.Fragment, null, (0, _element.createElement)("h1", {
        className: "edit-post-welcome-guide__heading"
      }, (0, _i18n.__)('Learn how to use the block editor')), (0, _element.createElement)("p", {
        className: "edit-post-welcome-guide__text"
      }, (0, _i18n.__)('New to the block editor? Want to learn more about using it? '), (0, _element.createElement)(_components.ExternalLink, {
        href: (0, _i18n.__)('https://wordpress.org/support/article/wordpress-editor/')
      }, (0, _i18n.__)("Here's a detailed guide."))))
    }]
  });
}
