// @wordpress/block-library/build-module/navigation/responsive-wrapper.js
import { createElement, Fragment } from "@wordpress/element";

/**
 * External dependencies
 */
import classnames from 'classnames';
/**
 * WordPress dependencies
 */

import { close, Icon } from '@wordpress/icons';
import { Button } from '@wordpress/components';
import { SVG, Rect } from '@wordpress/primitives';
import { __ } from '@wordpress/i18n';
export default function ResponsiveWrapper({
  children,
  id,
  isOpen,
  isResponsive,
  onToggle
}) {
  if (!isResponsive) {
    return children;
  }

  const responsiveContainerClasses = classnames('wp-block-navigation__responsive-container', {
    'is-menu-open': isOpen
  });
  const modalId = `${id}-modal`;
  return createElement(Fragment, null, !isOpen && createElement(Button, {
    "aria-haspopup": "true",
    "aria-expanded": isOpen,
    "aria-label": __('Open menu'),
    className: "wp-block-navigation__responsive-container-open",
    onClick: () => onToggle(true)
  }, createElement(SVG, {
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 24 24",
    width: "24",
    height: "24",
    role: "img",
    "aria-hidden": "true",
    focusable: "false"
  }, createElement(Rect, {
    x: "4",
    y: "7.5",
    width: "16",
    height: "1.5"
  }), createElement(Rect, {
    x: "4",
    y: "15",
    width: "16",
    height: "1.5"
  }))), createElement("div", {
    className: responsiveContainerClasses,
    id: modalId,
    "aria-hidden": !isOpen
  }, createElement("div", {
    className: "wp-block-navigation__responsive-close",
    tabIndex: "-1"
  }, createElement("div", {
    className: "wp-block-navigation__responsive-dialog",
    role: "dialog",
    "aria-modal": "true",
    "aria-labelledby": `${modalId}-title`
  }, createElement(Button, {
    className: "wp-block-navigation__responsive-container-close",
    "aria-label": __('Close menu'),
    onClick: () => onToggle(false)
  }, createElement(Icon, {
    icon: close
  })), createElement("div", {
    className: "wp-block-navigation__responsive-container-content",
    id: `${modalId}-content`
  }, children)))));
}
