// @wordpress/block-library/build-module/navigation/use-block-navigator.js
import { createElement } from "@wordpress/element";

/**
 * WordPress dependencies
 */
import { useState } from '@wordpress/element';
import { ToolbarButton, Modal } from '@wordpress/components';
import { __ } from '@wordpress/i18n';
import { listView } from '@wordpress/icons';
/**
 * Internal dependencies
 */

import BlockNavigationList from './block-navigation-list';
export default function useBlockNavigator(clientId, __experimentalFeatures) {
  const [isNavigationListOpen, setIsNavigationListOpen] = useState(false);
  const navigatorToolbarButton = createElement(ToolbarButton, {
    className: "components-toolbar__control",
    label: __('Open block navigation'),
    onClick: () => setIsNavigationListOpen(true),
    icon: listView
  });
  const navigatorModal = isNavigationListOpen && createElement(Modal, {
    title: __('Navigation'),
    closeLabel: __('Close'),
    onRequestClose: () => {
      setIsNavigationListOpen(false);
    },
    shouldCloseOnClickOutside: false
  }, createElement(BlockNavigationList, {
    clientId: clientId,
    __experimentalFeatures: __experimentalFeatures
  }));
  return {
    navigatorToolbarButton,
    navigatorModal
  };
}
