// @wordpress/edit-post/build/components/header/more-menu/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _element = require("@wordpress/element");

var _i18n = require("@wordpress/i18n");

var _components = require("@wordpress/components");

var _interface = require("@wordpress/interface");

var _compose = require("@wordpress/compose");

var _modeSwitcher = _interopRequireDefault(require("../mode-switcher"));

var _preferencesMenuItem = _interopRequireDefault(require("../preferences-menu-item"));

var _toolsMoreMenuGroup = _interopRequireDefault(require("../tools-more-menu-group"));

var _writingMenu = _interopRequireDefault(require("../writing-menu"));

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
const POPOVER_PROPS = {
  className: 'edit-post-more-menu__content'
};

const MoreMenu = ({
  showIconLabels
}) => {
  const isLargeViewport = (0, _compose.useViewportMatch)('large');
  return (0, _element.createElement)(_interface.MoreMenuDropdown, {
    className: "edit-post-more-menu",
    popoverProps: POPOVER_PROPS,
    toggleProps: {
      showTooltip: !showIconLabels,
      ...(showIconLabels && {
        variant: 'tertiary'
      })
    }
  }, ({
    onClose
  }) => (0, _element.createElement)(_element.Fragment, null, showIconLabels && !isLargeViewport && (0, _element.createElement)(_interface.PinnedItems.Slot, {
    className: showIconLabels && 'show-icon-labels',
    scope: "core/edit-post"
  }), (0, _element.createElement)(_writingMenu.default, null), (0, _element.createElement)(_modeSwitcher.default, null), (0, _element.createElement)(_interface.ActionItem.Slot, {
    name: "core/edit-post/plugin-more-menu",
    label: (0, _i18n.__)('Plugins'),
    as: _components.MenuGroup,
    fillProps: {
      onClick: onClose
    }
  }), (0, _element.createElement)(_toolsMoreMenuGroup.default.Slot, {
    fillProps: {
      onClose
    }
  }), (0, _element.createElement)(_components.MenuGroup, null, (0, _element.createElement)(_preferencesMenuItem.default, null))));
};

var _default = MoreMenu;
exports.default = _default;
