// @wordpress/block-library/build/table-of-contents/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.settings = exports.name = exports.metadata = void 0;

var _edit = _interopRequireDefault(require("./edit"));

var _icon = _interopRequireDefault(require("./icon"));

/**
 * Internal dependencies
 */
const metadata = {
  apiVersion: 2,
  name: "core/table-of-contents",
  title: "Table of Contents",
  category: "layout",
  description: "Summarize your post with a list of headings. Add HTML anchors to Heading blocks to link them here.",
  keywords: ["document outline", "summary"],
  textdomain: "default",
  attributes: {
    onlyIncludeCurrentPage: {
      type: "boolean",
      "default": false
    }
  },
  usesContext: ["postId"],
  supports: {
    html: false
  }
};
exports.metadata = metadata;
const {
  name
} = metadata;
exports.name = name;
const settings = {
  icon: _icon.default,
  edit: _edit.default
};
exports.settings = settings;
