// @wordpress/editor/build/components/editor-help/help-detail-navigation-screen.native.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _element = require("@wordpress/element");

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _reactNative = require("react-native");

var _native = require("@react-navigation/native");

var _components = require("@wordpress/components");

var _style = _interopRequireDefault(require("./style.scss"));

/**
 * External dependencies
 */

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
const HelpDetailNavigationScreen = ({
  content,
  label
}) => {
  const navigation = (0, _native.useNavigation)();
  const {
    listProps
  } = (0, _element.useContext)(_components.BottomSheetContext);

  const contentContainerStyle = _reactNative.StyleSheet.flatten(listProps.contentContainerStyle);

  return (0, _element.createElement)(_components.BottomSheet.NavigationScreen, {
    isScrollable: true,
    fullScreen: true
  }, (0, _element.createElement)(_reactNative.View, {
    style: _style.default.container
  }, (0, _element.createElement)(_components.BottomSheet.NavBar, null, (0, _element.createElement)(_components.BottomSheet.NavBar.BackButton, {
    onPress: navigation.goBack
  }), (0, _element.createElement)(_components.BottomSheet.NavBar.Heading, null, label)), (0, _element.createElement)(_reactNative.ScrollView, (0, _extends2.default)({}, listProps, {
    contentContainerStyle: { ...contentContainerStyle,
      paddingBottom: Math.max(listProps.safeAreaBottomInset, contentContainerStyle.paddingBottom),

      /**
       * Remove margin set via `hideHeader`. Combining a header
       * and navigation in this bottom sheet is at odds with the
       * current `BottomSheet` implementation.
       */
      marginTop: 0
    }
  }), (0, _element.createElement)(_reactNative.TouchableWithoutFeedback, {
    accessible: false
  }, (0, _element.createElement)(_reactNative.View, null, content)))));
};

var _default = HelpDetailNavigationScreen;
exports.default = _default;
