// @wordpress/block-library/build/post-terms/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.settings = exports.name = exports.metadata = void 0;

var _edit = _interopRequireDefault(require("./edit"));

var _variations = _interopRequireDefault(require("./variations"));

/**
 * Internal dependencies
 */
const metadata = {
  apiVersion: 2,
  name: "core/post-terms",
  title: "Post Terms",
  category: "design",
  description: "Post terms.",
  textdomain: "default",
  attributes: {
    term: {
      type: "string"
    },
    textAlign: {
      type: "string"
    },
    separator: {
      type: "string",
      "default": ", "
    }
  },
  usesContext: ["postId", "postType"],
  supports: {
    html: false,
    color: {
      gradients: true,
      link: true
    },
    typography: {
      lineHeight: true,
      fontSize: true,
      __experimentalFontWeight: true
    }
  },
  style: "wp-block-post-terms"
};
exports.metadata = metadata;
const {
  name
} = metadata;
exports.name = name;
const settings = {
  variations: _variations.default,
  edit: _edit.default
};
exports.settings = settings;
