// @wordpress/edit-post/build/components/preferences-modal/options/enable-custom-fields.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CustomFieldsConfirmation = CustomFieldsConfirmation;
exports.EnableCustomFieldsOption = EnableCustomFieldsOption;
exports.default = void 0;

var _element = require("@wordpress/element");

var _i18n = require("@wordpress/i18n");

var _components = require("@wordpress/components");

var _data = require("@wordpress/data");

var _editor = require("@wordpress/editor");

var _base = _interopRequireDefault(require("./base"));

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
function CustomFieldsConfirmation({
  willEnable
}) {
  const [isReloading, setIsReloading] = (0, _element.useState)(false);
  return (0, _element.createElement)(_element.Fragment, null, (0, _element.createElement)("p", {
    className: "edit-post-preferences-modal__custom-fields-confirmation-message"
  }, (0, _i18n.__)('A page reload is required for this change. Make sure your content is saved before reloading.')), (0, _element.createElement)(_components.Button, {
    className: "edit-post-preferences-modal__custom-fields-confirmation-button",
    variant: "secondary",
    isBusy: isReloading,
    disabled: isReloading,
    onClick: () => {
      setIsReloading(true);
      document.getElementById('toggle-custom-fields-form').submit();
    }
  }, willEnable ? (0, _i18n.__)('Enable & Reload') : (0, _i18n.__)('Disable & Reload')));
}

function EnableCustomFieldsOption({
  label,
  areCustomFieldsEnabled
}) {
  const [isChecked, setIsChecked] = (0, _element.useState)(areCustomFieldsEnabled);
  return (0, _element.createElement)(_base.default, {
    label: label,
    isChecked: isChecked,
    onChange: setIsChecked
  }, isChecked !== areCustomFieldsEnabled && (0, _element.createElement)(CustomFieldsConfirmation, {
    willEnable: isChecked
  }));
}

var _default = (0, _data.withSelect)(select => ({
  areCustomFieldsEnabled: !!select(_editor.store).getEditorSettings().enableCustomFields
}))(EnableCustomFieldsOption);

exports.default = _default;
