// @wordpress/block-library/build-module/site-logo/index.js
/**
 * WordPress dependencies
 */
import { siteLogo as icon } from '@wordpress/icons';
/**
 * Internal dependencies
 */

const metadata = {
  apiVersion: 2,
  name: "core/site-logo",
  title: "Site Logo",
  category: "layout",
  description: "Display a graphic to represent this site. Update the block, and the changes apply everywhere it\u2019s used. This is different than the site icon, which is the smaller image visible in your dashboard, browser tabs, etc used to help others recognize this site.",
  textdomain: "default",
  attributes: {
    align: {
      type: "string"
    },
    width: {
      type: "number"
    },
    isLink: {
      type: "boolean",
      "default": true
    },
    linkTarget: {
      type: "string",
      "default": "_self"
    }
  },
  supports: {
    html: false,
    align: true,
    alignWide: false,
    color: {
      __experimentalDuotone: "img",
      text: false,
      background: false
    }
  },
  styles: [{
    name: "default",
    label: "Default",
    isDefault: true
  }, {
    name: "rounded",
    label: "Rounded"
  }],
  editorStyle: "wp-block-site-logo-editor",
  style: "wp-block-site-logo"
};
import edit from './edit';
const {
  name
} = metadata;
export { metadata, name };
export const settings = {
  icon,
  edit
};
