// @wordpress/editor/build/components/post-sticky/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PostSticky = PostSticky;
exports.default = void 0;

var _element = require("@wordpress/element");

var _i18n = require("@wordpress/i18n");

var _components = require("@wordpress/components");

var _data = require("@wordpress/data");

var _compose = require("@wordpress/compose");

var _check = _interopRequireDefault(require("./check"));

var _store = require("../../store");

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
function PostSticky({
  onUpdateSticky,
  postSticky = false
}) {
  return (0, _element.createElement)(_check.default, null, (0, _element.createElement)(_components.CheckboxControl, {
    label: (0, _i18n.__)('Stick to the top of the blog'),
    checked: postSticky,
    onChange: () => onUpdateSticky(!postSticky)
  }));
}

var _default = (0, _compose.compose)([(0, _data.withSelect)(select => {
  return {
    postSticky: select(_store.store).getEditedPostAttribute('sticky')
  };
}), (0, _data.withDispatch)(dispatch => {
  return {
    onUpdateSticky(postSticky) {
      dispatch(_store.store).editPost({
        sticky: postSticky
      });
    }

  };
})])(PostSticky);

exports.default = _default;
