// @wordpress/editor/build/components/editor-notices/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EditorNotices = EditorNotices;
exports.default = void 0;

var _element = require("@wordpress/element");

var _lodash = require("lodash");

var _components = require("@wordpress/components");

var _data = require("@wordpress/data");

var _compose = require("@wordpress/compose");

var _notices = require("@wordpress/notices");

var _templateValidationNotice = _interopRequireDefault(require("../template-validation-notice"));

/**
 * External dependencies
 */

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
function EditorNotices({
  notices,
  onRemove
}) {
  const dismissibleNotices = (0, _lodash.filter)(notices, {
    isDismissible: true,
    type: 'default'
  });
  const nonDismissibleNotices = (0, _lodash.filter)(notices, {
    isDismissible: false,
    type: 'default'
  });
  return (0, _element.createElement)(_element.Fragment, null, (0, _element.createElement)(_components.NoticeList, {
    notices: nonDismissibleNotices,
    className: "components-editor-notices__pinned"
  }), (0, _element.createElement)(_components.NoticeList, {
    notices: dismissibleNotices,
    className: "components-editor-notices__dismissible",
    onRemove: onRemove
  }, (0, _element.createElement)(_templateValidationNotice.default, null)));
}

var _default = (0, _compose.compose)([(0, _data.withSelect)(select => ({
  notices: select(_notices.store).getNotices()
})), (0, _data.withDispatch)(dispatch => ({
  onRemove: dispatch(_notices.store).removeNotice
}))])(EditorNotices);

exports.default = _default;
