// @wordpress/interface/build/components/more-menu-dropdown/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = MoreMenuDropdown;

var _element = require("@wordpress/element");

var _classnames = _interopRequireDefault(require("classnames"));

var _components = require("@wordpress/components");

var _i18n = require("@wordpress/i18n");

var _icons = require("@wordpress/icons");

/**
 * External dependencies
 */

/**
 * WordPress dependencies
 */
function MoreMenuDropdown({
  as: DropdownComponent = _components.DropdownMenu,
  className,

  /* translators: button label text should, if possible, be under 16 characters. */
  label = (0, _i18n.__)('Options'),
  popoverProps,
  toggleProps,
  children
}) {
  return (0, _element.createElement)(DropdownComponent, {
    className: (0, _classnames.default)('interface-more-menu-dropdown', className),
    icon: _icons.moreVertical,
    label: label,
    popoverProps: {
      position: 'bottom left',
      ...popoverProps,
      className: (0, _classnames.default)('interface-more-menu-dropdown__content', popoverProps === null || popoverProps === void 0 ? void 0 : popoverProps.className)
    },
    toggleProps: {
      tooltipPosition: 'bottom',
      ...toggleProps
    }
  }, onClose => children(onClose));
}
