// @wordpress/block-library/build/navigation/block-navigation-list.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = BlockNavigationList;

var _element = require("@wordpress/element");

var _blockEditor = require("@wordpress/block-editor");

var _data = require("@wordpress/data");

/**
 * WordPress dependencies
 */
function BlockNavigationList({
  clientId,
  __experimentalFeatures
}) {
  const blocks = (0, _data.useSelect)(select => select(_blockEditor.store).__unstableGetClientIdsTree(clientId), [clientId]);
  const listViewRef = (0, _element.useRef)();
  const [minHeight, setMinHeight] = (0, _element.useState)(300);
  (0, _element.useEffect)(() => {
    var _listViewRef$current$, _listViewRef$current;

    setMinHeight((_listViewRef$current$ = listViewRef === null || listViewRef === void 0 ? void 0 : (_listViewRef$current = listViewRef.current) === null || _listViewRef$current === void 0 ? void 0 : _listViewRef$current.clientHeight) !== null && _listViewRef$current$ !== void 0 ? _listViewRef$current$ : 300);
  }, []);
  return (0, _element.createElement)("div", {
    style: {
      minHeight
    }
  }, (0, _element.createElement)(_blockEditor.__experimentalListView, {
    ref: listViewRef,
    blocks: blocks,
    showBlockMovers: true,
    showNestedBlocks: true,
    __experimentalFeatures: __experimentalFeatures
  }));
}
