// @wordpress/block-library/build/navigation/responsive-wrapper.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ResponsiveWrapper;

var _element = require("@wordpress/element");

var _classnames = _interopRequireDefault(require("classnames"));

var _icons = require("@wordpress/icons");

var _components = require("@wordpress/components");

var _primitives = require("@wordpress/primitives");

var _i18n = require("@wordpress/i18n");

/**
 * External dependencies
 */

/**
 * WordPress dependencies
 */
function ResponsiveWrapper({
  children,
  id,
  isOpen,
  isResponsive,
  onToggle
}) {
  if (!isResponsive) {
    return children;
  }

  const responsiveContainerClasses = (0, _classnames.default)('wp-block-navigation__responsive-container', {
    'is-menu-open': isOpen
  });
  const modalId = `${id}-modal`;
  return (0, _element.createElement)(_element.Fragment, null, !isOpen && (0, _element.createElement)(_components.Button, {
    "aria-haspopup": "true",
    "aria-expanded": isOpen,
    "aria-label": (0, _i18n.__)('Open menu'),
    className: "wp-block-navigation__responsive-container-open",
    onClick: () => onToggle(true)
  }, (0, _element.createElement)(_primitives.SVG, {
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 24 24",
    width: "24",
    height: "24",
    role: "img",
    "aria-hidden": "true",
    focusable: "false"
  }, (0, _element.createElement)(_primitives.Rect, {
    x: "4",
    y: "7.5",
    width: "16",
    height: "1.5"
  }), (0, _element.createElement)(_primitives.Rect, {
    x: "4",
    y: "15",
    width: "16",
    height: "1.5"
  }))), (0, _element.createElement)("div", {
    className: responsiveContainerClasses,
    id: modalId,
    "aria-hidden": !isOpen
  }, (0, _element.createElement)("div", {
    className: "wp-block-navigation__responsive-close",
    tabIndex: "-1"
  }, (0, _element.createElement)("div", {
    className: "wp-block-navigation__responsive-dialog",
    role: "dialog",
    "aria-modal": "true",
    "aria-labelledby": `${modalId}-title`
  }, (0, _element.createElement)(_components.Button, {
    className: "wp-block-navigation__responsive-container-close",
    "aria-label": (0, _i18n.__)('Close menu'),
    onClick: () => onToggle(false)
  }, (0, _element.createElement)(_icons.Icon, {
    icon: _icons.close
  })), (0, _element.createElement)("div", {
    className: "wp-block-navigation__responsive-container-content",
    id: `${modalId}-content`
  }, children)))));
}
