// @wordpress/block-library/build/navigation/use-block-navigator.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = useBlockNavigator;

var _element = require("@wordpress/element");

var _components = require("@wordpress/components");

var _i18n = require("@wordpress/i18n");

var _icons = require("@wordpress/icons");

var _blockNavigationList = _interopRequireDefault(require("./block-navigation-list"));

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
function useBlockNavigator(clientId, __experimentalFeatures) {
  const [isNavigationListOpen, setIsNavigationListOpen] = (0, _element.useState)(false);
  const navigatorToolbarButton = (0, _element.createElement)(_components.ToolbarButton, {
    className: "components-toolbar__control",
    label: (0, _i18n.__)('Open block navigation'),
    onClick: () => setIsNavigationListOpen(true),
    icon: _icons.listView
  });
  const navigatorModal = isNavigationListOpen && (0, _element.createElement)(_components.Modal, {
    title: (0, _i18n.__)('Navigation'),
    closeLabel: (0, _i18n.__)('Close'),
    onRequestClose: () => {
      setIsNavigationListOpen(false);
    },
    shouldCloseOnClickOutside: false
  }, (0, _element.createElement)(_blockNavigationList.default, {
    clientId: clientId,
    __experimentalFeatures: __experimentalFeatures
  }));
  return {
    navigatorToolbarButton,
    navigatorModal
  };
}
