// @wordpress/edit-post/build-module/components/sidebar/discussion-panel/index.js
import { createElement } from "@wordpress/element";

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { PanelBody, PanelRow } from '@wordpress/components';
import { PostComments, PostPingbacks, PostTypeSupportCheck } from '@wordpress/editor';
import { compose } from '@wordpress/compose';
import { withSelect, withDispatch } from '@wordpress/data';
/**
 * Internal dependencies
 */

import { store as editPostStore } from '../../../store';
/**
 * Module Constants
 */

const PANEL_NAME = 'discussion-panel';

function DiscussionPanel({
  isEnabled,
  isOpened,
  onTogglePanel
}) {
  if (!isEnabled) {
    return null;
  }

  return createElement(PostTypeSupportCheck, {
    supportKeys: ['comments', 'trackbacks']
  }, createElement(PanelBody, {
    title: __('Discussion'),
    opened: isOpened,
    onToggle: onTogglePanel
  }, createElement(PostTypeSupportCheck, {
    supportKeys: "comments"
  }, createElement(PanelRow, null, createElement(PostComments, null))), createElement(PostTypeSupportCheck, {
    supportKeys: "trackbacks"
  }, createElement(PanelRow, null, createElement(PostPingbacks, null)))));
}

export default compose([withSelect(select => {
  return {
    isEnabled: select(editPostStore).isEditorPanelEnabled(PANEL_NAME),
    isOpened: select(editPostStore).isEditorPanelOpened(PANEL_NAME)
  };
}), withDispatch(dispatch => ({
  onTogglePanel() {
    return dispatch(editPostStore).toggleEditorPanelOpened(PANEL_NAME);
  }

}))])(DiscussionPanel);
