// @wordpress/block-library/build-module/home-link/index.js
/**
 * WordPress dependencies
 */
import { _x } from '@wordpress/i18n';
import { home } from '@wordpress/icons';
/**
 * Internal dependencies
 */

const metadata = {
  apiVersion: 2,
  name: "core/home-link",
  category: "design",
  parent: ["core/navigation"],
  title: "Home Link",
  description: "Create a link that always points to the homepage of the site. Usually not necessary if there is already a site title link present in the header.",
  textdomain: "default",
  attributes: {
    label: {
      type: "string"
    }
  },
  usesContext: ["textColor", "customTextColor", "backgroundColor", "customBackgroundColor", "fontSize", "customFontSize", "style"],
  supports: {
    reusable: false,
    html: false
  },
  editorStyle: "wp-block-home-link-editor",
  style: "wp-block-home-link"
};
import edit from './edit';
import save from './save';
const {
  name
} = metadata;
export { metadata, name };
export const settings = {
  icon: home,
  edit,
  save,
  example: {
    attributes: {
      label: _x('Home Link', 'block example')
    }
  }
};
