// @wordpress/edit-post/build/components/sidebar/post-schedule/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PostSchedule = PostSchedule;
exports.default = void 0;

var _element = require("@wordpress/element");

var _i18n = require("@wordpress/i18n");

var _components = require("@wordpress/components");

var _editor = require("@wordpress/editor");

/**
 * WordPress dependencies
 */
function PostSchedule() {
  const anchorRef = (0, _element.useRef)();
  return (0, _element.createElement)(_editor.PostScheduleCheck, null, (0, _element.createElement)(_components.PanelRow, {
    className: "edit-post-post-schedule",
    ref: anchorRef
  }, (0, _element.createElement)("span", null, (0, _i18n.__)('Publish')), (0, _element.createElement)(_components.Dropdown, {
    popoverProps: {
      anchorRef: anchorRef.current
    },
    position: "bottom left",
    contentClassName: "edit-post-post-schedule__dialog",
    renderToggle: ({
      onToggle,
      isOpen
    }) => (0, _element.createElement)(_element.Fragment, null, (0, _element.createElement)(_components.Button, {
      className: "edit-post-post-schedule__toggle",
      onClick: onToggle,
      "aria-expanded": isOpen,
      variant: "tertiary"
    }, (0, _element.createElement)(_editor.PostScheduleLabel, null))),
    renderContent: () => (0, _element.createElement)(_editor.PostSchedule, null)
  })));
}

var _default = PostSchedule;
exports.default = _default;
