// @wordpress/edit-post/build/components/sidebar/post-status/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _element = require("@wordpress/element");

var _i18n = require("@wordpress/i18n");

var _components = require("@wordpress/components");

var _data = require("@wordpress/data");

var _compose = require("@wordpress/compose");

var _postVisibility = _interopRequireDefault(require("../post-visibility"));

var _postTrash = _interopRequireDefault(require("../post-trash"));

var _postSchedule = _interopRequireDefault(require("../post-schedule"));

var _postSticky = _interopRequireDefault(require("../post-sticky"));

var _postAuthor = _interopRequireDefault(require("../post-author"));

var _postSlug = _interopRequireDefault(require("../post-slug"));

var _postFormat = _interopRequireDefault(require("../post-format"));

var _postPendingStatus = _interopRequireDefault(require("../post-pending-status"));

var _pluginPostStatusInfo = _interopRequireDefault(require("../plugin-post-status-info"));

var _store = require("../../../store");

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */

/**
 * Module Constants
 */
const PANEL_NAME = 'post-status';

function PostStatus({
  isOpened,
  onTogglePanel
}) {
  return (0, _element.createElement)(_components.PanelBody, {
    className: "edit-post-post-status",
    title: (0, _i18n.__)('Status & visibility'),
    opened: isOpened,
    onToggle: onTogglePanel
  }, (0, _element.createElement)(_pluginPostStatusInfo.default.Slot, null, fills => (0, _element.createElement)(_element.Fragment, null, (0, _element.createElement)(_postVisibility.default, null), (0, _element.createElement)(_postSchedule.default, null), (0, _element.createElement)(_postFormat.default, null), (0, _element.createElement)(_postSticky.default, null), (0, _element.createElement)(_postPendingStatus.default, null), (0, _element.createElement)(_postSlug.default, null), (0, _element.createElement)(_postAuthor.default, null), fills, (0, _element.createElement)(_postTrash.default, null))));
}

var _default = (0, _compose.compose)([(0, _data.withSelect)(select => {
  // We use isEditorPanelRemoved to hide the panel if it was programatically removed. We do
  // not use isEditorPanelEnabled since this panel should not be disabled through the UI.
  const {
    isEditorPanelRemoved,
    isEditorPanelOpened
  } = select(_store.store);
  return {
    isRemoved: isEditorPanelRemoved(PANEL_NAME),
    isOpened: isEditorPanelOpened(PANEL_NAME)
  };
}), (0, _compose.ifCondition)(({
  isRemoved
}) => !isRemoved), (0, _data.withDispatch)(dispatch => ({
  onTogglePanel() {
    return dispatch(_store.store).toggleEditorPanelOpened(PANEL_NAME);
  }

}))])(PostStatus);

exports.default = _default;
