// @wordpress/components/build-module/toggle-group-control/component.js
import _extends from "@babel/runtime/helpers/esm/extends";
import { createElement } from "@wordpress/element";

/**
 * External dependencies
 */
// eslint-disable-next-line no-restricted-imports
import { RadioGroup, useRadioState } from 'reakit';
import useResizeAware from 'react-resize-aware';
/**
 * WordPress dependencies
 */

import { useRef, useMemo } from '@wordpress/element';
import { useMergeRefs, useInstanceId } from '@wordpress/compose';
/**
 * Internal dependencies
 */

import { contextConnect, useContextSystem } from '../ui/context';
import { useUpdateEffect, useCx } from '../utils/hooks';
import { View } from '../view';
import BaseControl from '../base-control';
import ToggleGroupControlBackdrop from './toggle-group-control-backdrop';
import ToggleGroupControlContext from './toggle-group-control-context';
import * as styles from './styles';

const noop = () => {};

function ToggleGroupControl(props, forwardedRef) {
  const {
    className,
    isAdaptiveWidth = false,
    isBlock = false,
    label,
    hideLabelFromVision = false,
    help,
    onChange = noop,
    value,
    children,
    ...otherProps
  } = useContextSystem(props, 'ToggleGroupControl');
  const cx = useCx();
  const containerRef = useRef();
  const [resizeListener, sizes] = useResizeAware();
  const baseId = useInstanceId(ToggleGroupControl, 'toggle-group-control').toString();
  const radio = useRadioState({
    baseId,
    state: value
  }); // Propagate radio.state change

  useUpdateEffect(() => {
    onChange(radio.state);
  }, [radio.state]); // Sync incoming value with radio.state

  useUpdateEffect(() => {
    if (value !== radio.state) {
      radio.setState(value);
    }
  }, [value]);
  const classes = useMemo(() => cx(styles.ToggleGroupControl, isBlock && styles.block, 'medium', className), [className, isBlock]);
  return createElement(BaseControl, {
    help: help
  }, createElement(ToggleGroupControlContext.Provider, {
    value: { ...radio,
      isBlock: !isAdaptiveWidth
    }
  }, !hideLabelFromVision && createElement("div", null, createElement(BaseControl.VisualLabel, null, label)), createElement(RadioGroup, _extends({}, radio, {
    "aria-label": label,
    as: View,
    className: classes
  }, otherProps, {
    ref: useMergeRefs([containerRef, forwardedRef])
  }), resizeListener, createElement(ToggleGroupControlBackdrop, _extends({}, radio, {
    containerRef: containerRef,
    containerWidth: sizes.width,
    isAdaptiveWidth: isAdaptiveWidth
  })), children)));
}

export default contextConnect(ToggleGroupControl, 'ToggleGroupControl');
