// @wordpress/interface/build/components/more-menu-feature-toggle/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = MoreMenuFeatureToggle;

var _element = require("@wordpress/element");

var _data = require("@wordpress/data");

var _components = require("@wordpress/components");

var _i18n = require("@wordpress/i18n");

var _icons = require("@wordpress/icons");

var _a11y = require("@wordpress/a11y");

var _store = require("../../store");

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
function MoreMenuFeatureToggle({
  scope,
  label,
  info,
  messageActivated,
  messageDeactivated,
  shortcut,
  feature
}) {
  const isActive = (0, _data.useSelect)(select => select(_store.store).isFeatureActive(scope, feature), [feature]);
  const {
    toggleFeature
  } = (0, _data.useDispatch)(_store.store);

  const speakMessage = () => {
    if (isActive) {
      (0, _a11y.speak)(messageDeactivated || (0, _i18n.__)('Feature deactivated'));
    } else {
      (0, _a11y.speak)(messageActivated || (0, _i18n.__)('Feature activated'));
    }
  };

  return (0, _element.createElement)(_components.MenuItem, {
    icon: isActive && _icons.check,
    isSelected: isActive,
    onClick: () => {
      toggleFeature(scope, feature);
      speakMessage();
    },
    role: "menuitemcheckbox",
    info: info,
    shortcut: shortcut
  }, label);
}
