// @wordpress/block-library/build-module/template-part/index.js
/**
 * External dependencies
 */
import { startCase } from 'lodash';
/**
 * WordPress dependencies
 */

import { store as coreDataStore } from '@wordpress/core-data';
import { select } from '@wordpress/data';
import { symbolFilled } from '@wordpress/icons';
import { addFilter } from '@wordpress/hooks';
/**
 * Internal dependencies
 */

const metadata = {
  apiVersion: 2,
  name: "core/template-part",
  title: "Template Part",
  category: "theme",
  description: "Edit the different global regions of your site, like the header, footer, sidebar, or create your own.",
  textdomain: "default",
  attributes: {
    slug: {
      type: "string"
    },
    theme: {
      type: "string"
    },
    tagName: {
      type: "string"
    },
    area: {
      type: "string"
    }
  },
  supports: {
    align: true,
    html: false,
    color: {
      gradients: true,
      link: true
    },
    spacing: {
      padding: true
    },
    __experimentalLayout: true
  },
  editorStyle: "wp-block-template-part-editor"
};
import edit from './edit';
import { enhanceTemplatePartVariations } from './variations';
const {
  name
} = metadata;
export { metadata, name };
export const settings = {
  icon: symbolFilled,
  __experimentalLabel: ({
    slug,
    theme
  }) => {
    var _entity$title;

    // Attempt to find entity title if block is a template part.
    // Require slug to request, otherwise entity is uncreated and will throw 404.
    if (!slug) {
      return;
    }

    const entity = select(coreDataStore).getEntityRecord('postType', 'wp_template_part', theme + '//' + slug);

    if (!entity) {
      return;
    }

    return startCase(((_entity$title = entity.title) === null || _entity$title === void 0 ? void 0 : _entity$title.rendered) || entity.slug);
  },
  edit
}; // Importing this file includes side effects. This is whitelisted in block-library/package.json under sideEffects

addFilter('blocks.registerBlockType', 'core/template-part', enhanceTemplatePartVariations);
