// @wordpress/edit-post/build/components/sidebar/post-visibility/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PostVisibility = PostVisibility;
exports.default = void 0;

var _element = require("@wordpress/element");

var _i18n = require("@wordpress/i18n");

var _components = require("@wordpress/components");

var _editor = require("@wordpress/editor");

/**
 * WordPress dependencies
 */
function PostVisibility() {
  return (0, _element.createElement)(_editor.PostVisibilityCheck, {
    render: ({
      canEdit
    }) => (0, _element.createElement)(_components.PanelRow, {
      className: "edit-post-post-visibility"
    }, (0, _element.createElement)("span", null, (0, _i18n.__)('Visibility')), !canEdit && (0, _element.createElement)("span", null, (0, _element.createElement)(_editor.PostVisibilityLabel, null)), canEdit && (0, _element.createElement)(_components.Dropdown, {
      position: "bottom left",
      contentClassName: "edit-post-post-visibility__dialog",
      renderToggle: ({
        isOpen,
        onToggle
      }) => (0, _element.createElement)(_components.Button, {
        "aria-expanded": isOpen,
        className: "edit-post-post-visibility__toggle",
        onClick: onToggle,
        variant: "tertiary"
      }, (0, _element.createElement)(_editor.PostVisibilityLabel, null)),
      renderContent: () => (0, _element.createElement)(_editor.PostVisibility, null)
    }))
  });
}

var _default = PostVisibility;
exports.default = _default;
