// @wordpress/edit-post/build-module/components/sidebar/post-visibility/index.js
import { createElement } from "@wordpress/element";

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { PanelRow, Dropdown, Button } from '@wordpress/components';
import { PostVisibility as PostVisibilityForm, PostVisibilityLabel, PostVisibilityCheck } from '@wordpress/editor';
export function PostVisibility() {
  return createElement(PostVisibilityCheck, {
    render: ({
      canEdit
    }) => createElement(PanelRow, {
      className: "edit-post-post-visibility"
    }, createElement("span", null, __('Visibility')), !canEdit && createElement("span", null, createElement(PostVisibilityLabel, null)), canEdit && createElement(Dropdown, {
      position: "bottom left",
      contentClassName: "edit-post-post-visibility__dialog",
      renderToggle: ({
        isOpen,
        onToggle
      }) => createElement(Button, {
        "aria-expanded": isOpen,
        className: "edit-post-post-visibility__toggle",
        onClick: onToggle,
        variant: "tertiary"
      }, createElement(PostVisibilityLabel, null)),
      renderContent: () => createElement(PostVisibilityForm, null)
    }))
  });
}
export default PostVisibility;
