// @wordpress/edit-post/build/components/sidebar/post-excerpt/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _element = require("@wordpress/element");

var _i18n = require("@wordpress/i18n");

var _components = require("@wordpress/components");

var _editor = require("@wordpress/editor");

var _compose = require("@wordpress/compose");

var _data = require("@wordpress/data");

var _store = require("../../../store");

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */

/**
 * Module Constants
 */
const PANEL_NAME = 'post-excerpt';

function PostExcerpt({
  isEnabled,
  isOpened,
  onTogglePanel
}) {
  if (!isEnabled) {
    return null;
  }

  return (0, _element.createElement)(_editor.PostExcerptCheck, null, (0, _element.createElement)(_components.PanelBody, {
    title: (0, _i18n.__)('Excerpt'),
    opened: isOpened,
    onToggle: onTogglePanel
  }, (0, _element.createElement)(_editor.PostExcerpt, null)));
}

var _default = (0, _compose.compose)([(0, _data.withSelect)(select => {
  return {
    isEnabled: select(_store.store).isEditorPanelEnabled(PANEL_NAME),
    isOpened: select(_store.store).isEditorPanelOpened(PANEL_NAME)
  };
}), (0, _data.withDispatch)(dispatch => ({
  onTogglePanel() {
    return dispatch(_store.store).toggleEditorPanelOpened(PANEL_NAME);
  }

}))])(PostExcerpt);

exports.default = _default;
