// @wordpress/components/build/toggle-group-control/component.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _element = require("@wordpress/element");

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _reakit = require("reakit");

var _reactResizeAware = _interopRequireDefault(require("react-resize-aware"));

var _compose = require("@wordpress/compose");

var _context = require("../ui/context");

var _hooks = require("../utils/hooks");

var _view = require("../view");

var _baseControl = _interopRequireDefault(require("../base-control"));

var _toggleGroupControlBackdrop = _interopRequireDefault(require("./toggle-group-control-backdrop"));

var _toggleGroupControlContext = _interopRequireDefault(require("./toggle-group-control-context"));

var styles = _interopRequireWildcard(require("./styles"));

/**
 * External dependencies
 */
// eslint-disable-next-line no-restricted-imports

/**
 * Internal dependencies
 */
const noop = () => {};

function ToggleGroupControl(props, forwardedRef) {
  const {
    className,
    isAdaptiveWidth = false,
    isBlock = false,
    label,
    hideLabelFromVision = false,
    help,
    onChange = noop,
    value,
    children,
    ...otherProps
  } = (0, _context.useContextSystem)(props, 'ToggleGroupControl');
  const cx = (0, _hooks.useCx)();
  const containerRef = (0, _element.useRef)();
  const [resizeListener, sizes] = (0, _reactResizeAware.default)();
  const baseId = (0, _compose.useInstanceId)(ToggleGroupControl, 'toggle-group-control').toString();
  const radio = (0, _reakit.useRadioState)({
    baseId,
    state: value
  }); // Propagate radio.state change

  (0, _hooks.useUpdateEffect)(() => {
    onChange(radio.state);
  }, [radio.state]); // Sync incoming value with radio.state

  (0, _hooks.useUpdateEffect)(() => {
    if (value !== radio.state) {
      radio.setState(value);
    }
  }, [value]);
  const classes = (0, _element.useMemo)(() => cx(styles.ToggleGroupControl, isBlock && styles.block, 'medium', className), [className, isBlock]);
  return (0, _element.createElement)(_baseControl.default, {
    help: help
  }, (0, _element.createElement)(_toggleGroupControlContext.default.Provider, {
    value: { ...radio,
      isBlock: !isAdaptiveWidth
    }
  }, !hideLabelFromVision && (0, _element.createElement)("div", null, (0, _element.createElement)(_baseControl.default.VisualLabel, null, label)), (0, _element.createElement)(_reakit.RadioGroup, (0, _extends2.default)({}, radio, {
    "aria-label": label,
    as: _view.View,
    className: classes
  }, otherProps, {
    ref: (0, _compose.useMergeRefs)([containerRef, forwardedRef])
  }), resizeListener, (0, _element.createElement)(_toggleGroupControlBackdrop.default, (0, _extends2.default)({}, radio, {
    containerRef: containerRef,
    containerWidth: sizes.width,
    isAdaptiveWidth: isAdaptiveWidth
  })), children)));
}

var _default = (0, _context.contextConnect)(ToggleGroupControl, 'ToggleGroupControl');

exports.default = _default;
