// @wordpress/editor/build/components/post-author/select.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _element = require("@wordpress/element");

var _i18n = require("@wordpress/i18n");

var _data = require("@wordpress/data");

var _htmlEntities = require("@wordpress/html-entities");

var _components = require("@wordpress/components");

var _coreData = require("@wordpress/core-data");

var _store = require("../../store");

var _constants = require("./constants");

/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */
function PostAuthorSelect() {
  const {
    editPost
  } = (0, _data.useDispatch)(_store.store);
  const {
    postAuthor,
    authors
  } = (0, _data.useSelect)(select => {
    return {
      postAuthor: select(_store.store).getEditedPostAttribute('author'),
      authors: select(_coreData.store).getUsers(_constants.AUTHORS_QUERY)
    };
  }, []);
  const authorOptions = (0, _element.useMemo)(() => {
    return (authors !== null && authors !== void 0 ? authors : []).map(author => {
      return {
        value: author.id,
        label: (0, _htmlEntities.decodeEntities)(author.name)
      };
    });
  }, [authors]);

  const setAuthorId = value => {
    const author = Number(value);
    editPost({
      author
    });
  };

  return (0, _element.createElement)(_components.SelectControl, {
    className: "post-author-selector",
    label: (0, _i18n.__)('Author'),
    options: authorOptions,
    onChange: setAuthorId,
    value: postAuthor
  });
}

var _default = PostAuthorSelect;
exports.default = _default;
